extensions/figma-files/src/menu-bar.tsx

```diff
@@ -0,0 +1,63 @@
+import { MenuBarExtra, open } from "@raycast/api";
+import { useProjectFiles } from "./hooks/useProjectFiles";
+import { useFigmaApp } from "./hooks/useFigmaApp";
+import { useVisitedFiles } from "./hooks/useVisitedFiles";
+
+export default function Command() {
+  const { projectFiles, isLoading: isLoadingProjectFiles, hasError } = useProjectFiles();
+  const { files: visitedFiles, visitFile, isLoading: isLoadingVisitedFiles } = useVisitedFiles();
+
+  const desktopApp = useFigmaApp();
+
+  let url = "figma://file/";
+  if (!desktopApp) {
+    url = "https://figma.com/file/";
+  }
+
+  return (
+    <MenuBarExtra
+      icon={{
+        source: {
+          light: "figma-menubar-icon-dark.png",
+          dark: "figma-menubar-icon-light.png",
+        },
+      }}
+      tooltip="Figma files"
```

Comment:
Right now, every time the menu bar extra is opened, there's a bit of flickering; this is because the `visitedFiles` and/or `projectFiles` are initially `undefined` / empty, which render the menu as empty, only to then immediately render again, once the real data has come back.

To get around this, you should bass `isLoading` to `MenuBarExtra`. When `isLoading` is true, Raycast will skip empty or identical renders; once all the data has come back and `isLoading` is set to false, Raycast will render whatever the extension sends (including an empty menu).

```suggestion
      tooltip="Figma files"
      isLoading={isLoadingVisitedFiles || isLoadingProjectFiles}
```